### DISCUSSION TITLE
Linking to Branches from Issues lists only around 50 Branches, even when searching

### DISCUSSION THREAD

**Discussion by rbability** [community/community#61818] [Projects and Issues]
### Select Topic Area

Product Feedback

### Feature Area

Issues

### Body

I got this feedback from our Developers, so I hope I describe it correctly.

From within an issue, on the right side you have the opportunity to either create a new Branch or to link an existing branch. When you click on that, you can select a Repo that includes the branch and then get a scrollable list of all active(?) branches from this repo. This list contains around 50 entries and then it stopps. If you enter something in the search field, our developers expected that they can find their branch even if it is not within the list of the ~50, but the search seems to filter only on the 50 displayed already.
It would be great if the search would actually search on all branches of the repo and not only the ones listed... or shows more than 50 :-)

I proposed that they create their branches from within the issues, but some workflows require them to create issues + branches in separate tools / steps, so this is not always an option.

### Guidelines

- [x] I have read the above statement and can confirm my post is relevant to the GitHub feature areas [Issues](https://docs.github.com/en/issues/tracking-your-work-with-issues/about-issues) and/or [Projects](https://docs.github.com/en/issues/planning-and-tracking-with-projects/learning-about-projects/about-projects).

**Comment by kozmaz87** [comment]
To me this only lists Pull Requests. Branches can't be selected at all and when searching for them nothing shows up.

**Comment by rbability** [comment]
That is interesting @kozmaz87. We are on GitHub Enterprise Cloud, maybe there is differenb behaviour depending on the licensing?
